src/lib/decorations/path.typ

```diff
@@ -59,6 +63,16 @@
   factor: 150%,
 )
 
+#let resolve-amplitude(amplitude, segment, num-segments) = {
+  return if type(amplitude) == function {
+    (amplitude)(segment / (num-segments - 1) * 100%)
```

Comment:
This now needs to be reverted back to `(amplitude)(segment / num-segments * 100%)`, since `segment` now ranges from `0.25` to `num-segments + 0.75` 😅

You can see what I mean with the test case
```typ
wave(line((0,3), (4,3)), amplitude: t => { calc.sin(t/100%*calc.pi) })
```

Reply:
Yeah, I am playing around with it. With the .25 addition, you won't get `0%` in the callback, which is a bit unexpected, I think.

Reply:
I guess the `segment` variable is no longer an index, but rather the coordinate along the curve `0 <= t <= 1`